server/app/views/admin/programs/ProgramFormBuilder.java

```diff
@@ -89,26 +109,38 @@ private static FormTag buildProgramForm(
                     .setValue(DisplayMode.HIDDEN_IN_INDEX.getValue())
                     .setChecked(displayMode.equals(DisplayMode.HIDDEN_IN_INDEX.getValue()))
                     .getRadioTag()),
-        h2("Visible to applicants").withClasses(Styles.PY_2),
-        FieldWithLabel.input()
-            .setId("program-display-name-input")
-            .setFieldName("localizedDisplayName")
-            .setLabelText("Enter the publicly displayed name for this program*")
-            .setValue(displayName)
-            .getInputTag(),
         FieldWithLabel.textArea()
-            .setId("program-display-description-textarea")
-            .setFieldName("localizedDisplayDescription")
-            .setLabelText("Describe this program for the public*")
-            .setValue(displayDescription)
+            .setId("program-description-textarea")
+            .setFieldName("adminDescription")
+            .setLabelText("Program note for administrative use only*")
+            .setValue(adminDescription)
             .getTextareaTag(),
-        FieldWithLabel.input()
-            .setId("program-external-link-input")
-            .setFieldName("externalLink")
-            .setLabelText("Link to program website")
-            .setValue(externalLink)
-            .getInputTag(),
         submitButton("Save").withId("program-update-button"));
     return formTag;
   }
+
+  private DomContent programUrlField(String adminName, boolean editExistingProgram) {
+    if (editExistingProgram) {
+      String programUrl =
+          baseUrl
+              + controllers.applicant.routes.RedirectController.programBySlug(
+                      MainModule.SLUGIFIER.slugify(adminName))
+                  .url();
+      return div()
+          .withClass(Styles.MB_2)
+          .with(
+              input().isHidden().withName("adminName").withValue(adminName),
```

Comment:
We need to render it here just to later updates happy as they expect that field to be present? And ideally we wouldn't need to render and send it back given that we know it's immutable, right?